pandas/tests/io/formats/test_printing.py

```diff
@@ -207,3 +208,24 @@ def test_multiindex_long_element():
         "cccccccccccccccccccccc',)],\n           )"
     )
     assert str(data) == expected
+
+
+@pytest.mark.parametrize(
+    "data,output",
+    [
+        ([2, complex("nan"), 1], ["2.0+0.0j", "NaN+0.0j", "1.0+0.0j"]),
+        ([1.23j, complex("nan"), 1.23], ["0.00+1.23j", " NaN+0.00j", "1.23+0.00j"]),
```

Comment:
Could you add two more cases one with all negative numbers and complex nan and one with positive & negative elements and complex nan?

Reply:
Will do ASAP.